Extend install doc to cover nginx and improve nginx config (#5232)
os-autoinst/openQA: docs/Installing.asciidoc

This will automatically setup a local apache http proxy. Read on for more
detailed setup instructions with all the details.

> It is required to run openQA behind an http proxy (apache, nginx, etc..). See the

Reviewer: I don't think it's actually required. We should reword this. How about

```
- It is required to run openQA … . See the
+ If you wish to run openQA … then see the

> If you wish to run openQA behind an http proxy (apache, NGINX, etc..) then see the

Reviewer 2: ```suggestion
If you wish to run openQA behind an HTTP proxy (Apache, NGINX, …) then see the
```
("etc" is usually just followed by one dot. However, I suggest you simply use "…" instead of "etc".)

*openqa.conf.template* config file in */etc/apache2/vhosts.d* (openSUSE) or
`/etc/httpd/conf.d` (Fedora) when using apache2 or the config file in

> `/etc/nginx/conf.d` for nginx.

Reviewer: you supplied it in vhosts.d though for nginx altough I am not sure this is correct.

> `/etc/nginx/vhosts.d` for nginx.

Reviewer 2: ```suggestion
`/etc/nginx/vhosts.d` for NGINX.
```
or
```suggestion
`/etc/nginx/vhosts.d` for Nginx.
```
It doesn't really matter which one; just be consistent with the spelling one sentence before.

=== Apache proxy

> To make everything work correctly on openSUSE, you

Reviewer: ```suggestion
To make everything work correctly on openSUSE when using apache, you
```

> To make everything work correctly on openSUSE when using apache, you

Reviewer 2: ```suggestion
To make everything work correctly on openSUSE when using Apache, you
```

This will direct all HTTP traffic to openQA.

[source,sh]
--------------------------------------------------------------------------------
cp /etc/apache2/vhosts.d/openqa.conf.template /etc/apache2/vhosts.d/openqa.conf
--------------------------------------------------------------------------------

=== NGINX proxy

For a basic setup, you can copy *openqa.conf.template* to *openqa.conf*
and modify the `server_name` setting if required.
This will direct all HTTP traffic to openQA.

[source,sh]
--------------------------------------------------------------------------------
cp /etc/nginx/vhosts.d/openqa.conf.template /etc/nginx/vhosts.d/openqa.conf
--------------------------------------------------------------------------------

Note that the default config in *openqa.conf.template* is using the keyword
`default_server` in the `listen` statement. This will only change the behaviour
when accessing the server via its IP address. This means that the default vhost
for `localhost` in *nginx.conf* will take precedence when accessing the server
via `localhost`. You might want to disable it.

For openQA you need to set `httpsonly = 0` as described in the TLS/SSL section

> below, if you don't setup nginx for ssl.

Reviewer 2: ```suggestion
below, if you do not setup NGINX for SSL.
```

cp /etc/apache2/vhosts.d/openqa.conf.template /etc/apache2/vhosts.d/openqa.conf
--------------------------------------------------------------------------------

> === NGINX proxy

Reviewer: https://nginx.org/en/ writes it with lowercase so I would use `nginx proxy` here as well

The nginx config is already provided by the openQA RPM package.

> You only need to start the nginx server.

Reviewer: That I would like to see a proof for because https://progress.opensuse.org/issues/131024#note-11 showed me something different :)
